# Tarea 02 — Carga de Datos con Pandas

**Instrucciones:** Completa cada celda de código con tu respuesta.  
Usarás el dataset de Iris disponible en la URL indicada.

In [1]:
import pandas as pd

---
## Parte 1 — Cargar y explorar datos

**P1.** Carga el dataset de Iris desde la siguiente URL y guárdalo en una variable llamada `df`.  
URL: `https://raw.githubusercontent.com/mwaskom/seaborn-data/master/iris.csv`

In [3]:
url = "https://raw.githubusercontent.com/mwaskom/seaborn-data/master/iris.csv"

# Tu respuesta aquí
df = pd.read_csv(url)

**P2.** Muestra las primeras 5 filas del DataFrame.

In [4]:
# Tu respuesta aquí
df.head(5)

,sepal_length,sepal_width,petal_length,petal_width,species
0,5.1,3.5,1.4,0.2,setosa
1,4.9,3.0,1.4,0.2,setosa
2,4.7,3.2,1.3,0.2,setosa
3,4.6,3.1,1.5,0.2,setosa
4,5.0,3.6,1.4,0.2,setosa


**P3.** Usa `.info()` para ver los tipos de columnas y si hay valores nulos.  
Luego usa `.describe()` para ver el resumen estadístico. ¿Cuántas filas tiene el dataset?

In [5]:
# Tu respuesta aquí
df.info()
df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal_length  150 non-null    float64
 1   sepal_width   150 non-null    float64
 2   petal_length  150 non-null    float64
 3   petal_width   150 non-null    float64
 4   species       150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


,sepal_length,sepal_width,petal_length,petal_width
count,150.000000,150.000000,150.000000,150.000000
mean,5.843333,3.057333,3.758000,1.199333
std,0.828066,0.435866,1.765298,0.762238
min,4.300000,2.000000,1.000000,0.100000
25%,5.100000,2.800000,1.600000,0.300000
50%,5.800000,3.000000,4.350000,1.300000
75%,6.400000,3.300000,5.100000,1.800000
max,7.900000,4.400000,6.900000,2.500000


**Respuesta:** El dataset tiene **150** filas.

> Añadir blockquote



---
## Parte 2 — Selección y filtrado

**P4.** Selecciona solo las columnas `sepal_length` y `species` e imprime las primeras 5 filas.

In [6]:
# Tu respuesta aquí
df[["sepal_length", "species"]].head(5)


,sepal_length,species
0,5.1,setosa
1,4.9,setosa
2,4.7,setosa
3,4.6,setosa
4,5.0,setosa


**P5.** Filtra el DataFrame para quedarte solo con las flores de la especie `virginica`.  
¿Cuántas filas tiene ese subconjunto?

In [7]:
# Tu respuesta aquí
virginica = df[df["species"] == "virginica"]
virginica.shape[0]


50

**P6.** Filtra las filas donde `petal_length` sea mayor a 5.0. Imprime cuántas filas cumplen esa condición.

In [12]:
# Tu respuesta aquí
df[df["petal_length"] > 5.0].shape[0]

42

---
## Parte 3 — Guardar datos

**P7.** Guarda el subconjunto de `virginica` (de P5) en un archivo CSV llamado `virginica.csv` en la carpeta `output/`.  
No incluyas el índice en el archivo guardado.

In [15]:
import os

# Tu respuesta aquí
os.makedirs("output", exist_ok=True)
virginica.to_csv("output/virginica.csv", index=False)

**P8.** Carga de nuevo el archivo `output/virginica.csv` y verifica que tiene el número correcto de filas usando `.shape`.

In [16]:
# Tu respuesta aquí
pd.read_csv("output/virginica.csv").shape[0]


50